examples/integration-tutorials/sending-a-simple-message-to-a-datasource/sql-data-service/db-interaction/data_service.bal

```diff
@@ -0,0 +1,227 @@
+            string payload = "json";
+
+            var retValue = addDoctorDetails(docName, hospital, speciality, availability, charge);
+            
+            if (retValue is sql:UpdateResult) {
+                if (retValue.updatedRowCount > 0) {
+                   
+                    json msg = "Insert successful";
+                    int statusCode = 200;
+
+                   var res = handleUpdate(caller, msg, statusCode, payload);
+                } else {
+
+                    json msg = "Insert Failed";
+                    int statusCode = 400;
+
+                    var res = handleUpdate(caller,msg, statusCode, payload);
+                }
+            } else {
+
+                json msg = "Insert Failed due to query execution failure";
+                int statusCode = 400;
+
+                var res = handleUpdate(caller,msg, statusCode, payload);
+            }
+        } else {
+            log:printError("Payload type does not match");
+            json msg = "Request payload is not json";
+            int statusCode = 400;
+            string payload = "other"; 
+
+            var res = handleUpdate(caller,msg, statusCode, payload);
+        }
+    }
+
+    // Resource that handles the HTTP PUT requests that are directed to the path '/doctor/<docName>' to update an 
+    // existing doctor's information.
+    @http:ResourceConfig {
+        methods: ["PUT"],
+        path: "/doctor/{docName}"
+    }
+    resource function updateDoctorData(http:Caller caller, http:Request request, string docName) {
+
+        http:Response response = new;
+
+        log:printInfo("Updating Doctors' Information");
+
+        var requestData = request.getJsonPayload();
+        
+        if (requestData is json) {
+            string hospital = <string> requestData.hospital;
+            string speciality = <string> requestData.speciality;
+            string availability = <string> requestData.availability;
+            int charge = <int> requestData.charge;
+            string payload = "json";
+
+            var retValue = updateDoctorDetails(docName, hospital, speciality, availability, charge);
+
+            if (retValue is sql:UpdateResult) {
+                if (retValue.updatedRowCount > 0) {
+                    json msg = "Update successful";
+                    int statusCode = 200;
+                    var res = handleUpdate(caller, msg, statusCode, payload);
+                } else {
+                    json msg = "Update Failed due to no matching record";
+                    int statusCode = 400;
+
+                    var res = handleUpdate(caller, msg, statusCode, payload);
+                }
+            } else {
+                json msg = "Update Failed while query execution";
+                int statusCode = 400;
+
+                var res = handleUpdate(caller, msg, statusCode, payload);
+            }      
+        } else {
+            log:printError("Payload type does not match");
```

Comment:
Shall we move error handling logic to a separate function?

Reply:
Fixed